### ISSUE TITLE
Will there be an example PHP client?

### ISSUE THREAD

**Issue by pies** [openraildata/stomp-client-php#1]

**Comment by ihateonions-uk** [comment]
just like the rail network, i guess not. probably running late.

**Comment by ellenby** [comment]
What exactly are you looking for it to do? There is example code that others including myself have put up on the NRODwiki... You could always write one yourself. I guess the issue being that everyone wants/needs the data for different things, so really should be interpreted individually.

**Comment by ellenby** [comment]
I could maybe put something basic together, but I don't really know what it is you're looking for. You basically need to connect to the STOMP server, and then with the nationalrail feeds you need to use a gz command to decode the message (not needed for network rail). The messages are then in (I believe) a JSON array, which you can interpret using json commands in much the same way as interpreting arrays in PHP normally.

**Comment by pies** [comment]
@ellenby Which function do I use to decode the message body? I've tried gzinflate, gzdecode, gzuncompress, zlib_decode and bzdecompress. All return errors.

**Comment by ellenby** [comment]
A simple example:

```
while($con && $i < $number){
   if ($con->hasFrame()){
       $msg = $con->readFrame();
       $event = gzdecode($msg->body)
       $i++ ;
         // do stuff with $event here
       $con->ack($msg);
   }
}
```

You have to decode the message body using gzdecode($msg->body)

The function gzdecode is also only available from PHP 5.4. If you are using an older version of PHP, upgrade.

See: http://nrodwiki.rockshore.net/index.php/PHP_Examples

Only the National Rail data needs decoding. The Network Rail data isn't coded so you don't need any gzdecode command.
